### PR TITLE
Prevent AWS credentials refresh from stopping on exception

### PR DESCRIPTION
**Pull request by aYukiSekiguchi**
If `aws_credentials()` fails due to an unstable network or other issues, it throws an exception. This stops `timer_execute()` from repeating its block, preventing `OpenSearchOutput` from updating `@_aws_credentials`. As a result, `@_aws_credentials` will expire.

This commit catches the exception and prevents it from propagating to `timer_execute()`, ensuring continuous credential updates.

### CODE REVIEW

**Comment by aYukiSekiguchi** [comment]
This will fix https://github.com/fluent/fluent-plugin-opensearch/issues/129

It looks like there is no test case for `@_aws_credentials` and `aws_credentials()`. I don't create a testcase for the exception. Sorry.

**Review by cosmo0920 — APPROVED** [review]

**Comment by akhil31415** [comment]
@aYukiSekiguchi san, I really appreciate the PR. Could you please check and fix the failing test cases? I kind of waiting for this to be merged.

**Comment by aYukiSekiguchi** [comment]
@akhil31415

I'm not very familiar with GitHub Actions, so I might be mistaken.

### Ruby 3.2 Unit Testing on ubuntu-latest
* [Failed Job Logs](https://github.com/fluent/fluent-plugin-opensearch/actions/runs/10667120322/job/29575733690#logs)
  * The job failed with the following error:
    ```
    The installation path is insecure. Bundler cannot continue.
    /opt/hostedtoolcache/Ruby/3.2.5/x64/lib/ruby/gems/3.2.0/gems is world-writable (without sticky bit).
    Bundler cannot safely replace gems in world-writable directories due to potential vulnerabilities.
    Please change the permissions of this directory or choose a different install path.
    Error: Process completed with exit code 38.
    ```
  * It seems there is an issue with the GitHub Action settings.
  * This might be resolved by re-running the action, but it appears I don't have the permissions to do so.

### Coverage/Coveralls
* As mentioned in my previous [comment](https://github.com/fluent/fluent-plugin-opensearch/pull/142#issuecomment-2324648275), there is no test for `@_aws_credentials`.
* Since I added `begin` / `rescue` to the `@_aws_credentials` code, this has decreased the branch coverage.
* I'm unsure if this project allows for a decrease in coverage.
* If decreasing coverage is not acceptable, I will try to write the necessary tests when I have some free time.

**Comment by ashie** [comment]
Sorry for our less activity in this plugin.
We are queuing this as our next task, will tackle on it next week.

**Comment by aYukiSekiguchi** [comment]
Thank you for retrying. It looks like there is a bug in `.github/workflows/linux.yml`. I have submitted a PR to fix the issue: [PR #143](https://github.com/fluent/fluent-plugin-opensearch/pull/143).

**Comment by cosmo0920** [comment]
Hi, I merged you fixing Linux workflow PR. Could you rebase off master?

**Comment by ntopee** [comment]
Sorry to rush, but is there any ETA on merging/releasing this change?
Thanks!

**Comment by cosmo0920** [comment]
@ashie @kenhys @daipom Could y'all take over this PR?

**Comment by aYukiSekiguchi** [comment]
Sorry for any confusion.

I'm waiting for the committer to decide whether the Coveralls failure, which decreased coverage, is acceptable. Some recent commits to the main branch also failed Coveralls, so I think this is sometimes acceptable.

**Comment by kenhys** [comment]
It seems that PR itself is reasonable but it might be better to add test case for it in assured.
I'll look into it.

**Comment by kenhys** [comment]
MEMO: it is enough that mock aws_credentials and raise RuntimeError in second call, then assert existence of error log message.

File: test/plugin/test_out_opensearch.rb
Comment (daipom) [inline test/plugin/test_out_opensearch.rb:4001]: * Power assert would be better for debug.
* `Array#any?` would be better for this purpose.

```suggestion
    assert { d.logs.any?(/\[error\]: Failed to get new AWS credentials: No valid AWS credentials found.\n/) }
```

File: test/plugin/test_out_opensearch.rb
Comment (daipom) [inline test/plugin/test_out_opensearch.rb:4000]: Is there any reason why you do not use `Driver#run`?

**Review by daipom — COMMENTED** [review]
Thanks!

File: test/plugin/test_out_opensearch.rb
Comment (daipom) [reply test/plugin/test_out_opensearch.rb:4000]: Is there any problem doing simply this?

```suggestion
    d.run
```

**Review by daipom — APPROVED** [review]
LGTM.

**Review by cosmo0920 — APPROVED** [review]

**Pull request merged**
